jaxrs-api/src/main/java/javax/ws/rs/JAXRS.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright (c) 2018 Markus KARG. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package javax.ws.rs;
+
+import javax.ws.rs.core.Application;
+import javax.ws.rs.ext.RuntimeDelegate;
+
+/**
+ * Bootstrap class that is used to startup a JAX-RS application in Java SE
+ * environments.
+ * <p>
+ * The {@code JAXRS} class is available in a Jakarta EE container environment as
+ * well; however, support for the Java SE bootstrapping APIs is <em>not
+ * required</em> in container environments.
+ * </p>
+ * <p>
+ * In a Java SE environment an application is getting started by the following
+ * command using default configuration values (i. e. mounting application at
+ * {@code http://localhost:80/}:
```

Comment:
On the other hand, virtually all public services run on 80 by default, as it is the default port of HTTP. 8080 is a workaround on Linux that I do not need on Windows. So shall we default on Linux-workarounds or wouldn't it be better to offload that decision to the application programmer? I mean, that's why a default that can be overridden easily.

Reply:
Where is the line between what has to and could be provided for the container to be started?

I mean.. host, port.. makes sense, but it doesn't need to be here, right? (host can be localhost by default and port can be any open port from some defined range - shouldn't that be also supported?)

what about thread pool sizes and threading model in general?
what about https, http/2 and related configuration?

Shouldn't we create examples (see https://github.com/eclipse-ee4j/jaxrs-api/tree/master/examples/src/main/java/jaxrs/examples) as it used to be done? (https is a standard requirement and should be part of that example/spec itself if this api should be portable)

Reply:
I agree with @chkal - `80` doesn't seem to be a good default - it won't run on many environments.

shouldn't there be a functionality which would pick any open port from defined range? Also, the instance should have getters for everything (?) which was passed as a configuration - and extendable part for container to talk back to user of this api - it could expose statistics and other advanced runtime info that way. (that would also solve the issue of getting the "randomly" selected port).

Reply:
* I also think 8080 would be a better default. However is less of an issue today with containers.
* I'd leave dynamic port selection to Docker, K8s, etc. if necessary
* +1 on getters

Reply:
So JAX-RS will state that anything other than docker container env is irrelevant for this API?

Getters would be useful to actually GET the configuration - I can image having some library which would work with an Instance and let's say I want to mention in the log I have received it - what would I get from it? There is no name, id or anything I could use to make the message good (not saying useful).

You could mention that if I want to do that, I should store the builder/configuration I used to create the instance - but why would I do that? Especially when Instance represents running instance, which does have some properties which clearly are managed by that class but are not exposes at this moment.

Reply:
@pavelbucek I don't think that by not supporting dynamic port ranges we are saying Docker is the only relevant environment. What I think we are saying is that it is and will be a very popular environment by the time we finish with this version of JAX-RS and it already solves that problem. A Bootstrap API like this would be very desirable in that very same environment.

As I said, I like the idea of getters if anything for completeness sake. It is a small and trivial addition too.

Reply:
Sure, the dynamic port range was mainly point for having getters, I'm not so keen of having something like that defined in the API. On the other hand, it could be nice to still have the possibility to implement something like that - which should be possible even now, since it can be defined via custom properties.

Reply:
Regarding ports: I don't think that dynamic port ranges make much sense. This shouldn't be the responsibility of JAX-RS. Regarding the default port: Of course port 80 is the default port for HTTP as defined by IANA, but I don't know a single Java EE container which defaults to this port. They all use 8080 and other non-privileged ports by default.

Reply:
Yes all Java EE containers start on 8080 because they are relicts from an ancient time where you had a full-blown OS underneath and a full-blown EE server. The idea behind this new API is to *not* have both, but to write a small Java SE `main()` that simply pulls up JAX-RS standalone in a micro-environment (such like docker). In such environments you do not have the mentioned restriction, as systems like docker map ports and users IDs automatically. So even when using root in the container you will not be root but a normal user actually, so even when using 80 in the container you will not be on 80 but anything beyond 1K actually.

Anyways, I do not want to stand in the way, but those not wanting 80 must now answer the question: What port to use when protocol is set up HTTPS? Not 443 I assume.

Reply:
Well, I wouldn't call it ancient to install a Java EE server on a server without wrapping it in a container. Maybe containers are the future, but non-container deployments aren't the past (yet). IMO one of the big advantages of being able start a JAX-RS runtime from a `main()` method is that you can start your server directly from the IDE without the need to set up anything. And this would fail with a default port of `80` (maybe except for Windows?). So defaulting to non-privileged ports still makes a lot of sense. BTW: IANA defines `8080` as an alternative HTTP port. So it is absolutely OK to use it from this perspective.

Regarding TLS: I think that using `8443` is a de-facto standard for non-privileged TLS ports.

Reply:
+1 on the use of 8080 and 8443
From an academic perspective there is a long history of recognizing 8080 and 8443 especially on a Windows platform for Java server/services access. We run servers that support IIS/Apache and GlassFish simultaneously for teaching purposes. The same holds true for our Linux servers.